hathor/p2p/states/ready.py

```diff
@@ -73,16 +70,13 @@ def __init__(self, protocol: 'HathorProtocol') -> None:
         # Initialize sync manager and add its commands to the list of available commands.
         connections = self.protocol.connections
         assert connections is not None
-        sync_factory: SyncManagerFactory
-        if self.protocol.protocol_version is ProtocolVersion.V1:
-            self.log.debug('loading sync-v1')
-            sync_factory = connections.sync_v1_factory
-        elif self.protocol.protocol_version is ProtocolVersion.V2:
-            self.log.debug('loading fake sync-v2 (actually sync-v1)')
-            sync_factory = connections.sync_v2_factory
-        else:
-            raise HathorError(f'wrong protocol version: {self.protocol.protocol_version}')
-        self.sync_manager: SyncManager = sync_factory.create_sync_manager(self.protocol, reactor=self.reactor)
+
+        sync_version = self.protocol.sync_version
+        self.log.debug(f'loading {sync_version}')
+        if sync_version not in connections.sync_factories:
+            raise HathorError(f'wrong protocol version: {self.protocol.sync_version}')
```

Comment:
Why raise an exception? Should we do `self.send_error_and_close_connection(...)` instead?

Or we can create a `connections.get_sync_factory(...)` method that will raise the exception, and here we will capture the exception and close the connection gracefully.

Reply:
I like `connections.get_sync_factory(...)` better. But this should be a harder error than just a `self.send_error_and_close_connection(...)`. I'd say this should be an assert (at the very least `assert self.protocol.sync_version is not None`, because we negotiated `self.protocol.sync_version` earlier and it shouldn't be possible to negotiate a version we don't have.

Reply:
Done.

Reply:
Oh! Ok. It seems using an assert is the best way to go.